include/irr/asset/normal_quantization.h

```diff
@@ -144,11 +171,16 @@ namespace asset
 	{
         QuantizationCacheEntry2_10_10_10 dummySearchVal;
         dummySearchVal.key = normal;
-        auto found = std::lower_bound(normalCacheFor2_10_10_10Quant.begin(),normalCacheFor2_10_10_10Quant.end(),dummySearchVal);
+        /*auto found = std::lower_bound(normalCacheFor2_10_10_10Quant.begin(),normalCacheFor2_10_10_10Quant.end(),dummySearchVal);
         if (found!=normalCacheFor2_10_10_10Quant.end()&&(found->key==normal).all())
         {
             return found->value;
-        }
+        }*/
+
+		auto found = normalCacheFor2_10_10_10QuantUm.find(normal);
```

Comment:
First of all use `fittingVector` not `normal`, we save space by noting the symmetry and using a `abs(normal)` for the fitting and caching (after fit is found or computed the function should restore the signs).

Should be applied to the search result too now
https://github.com/buildaworldnet/IrrlichtBAW/pull/437/files#diff-6e17f886f5c5ef84f97cb8acf20c9926R188

you want to take the `fittingVector` and transform it into a 2d surface projection `projFitting`

my suggestion is barycentric coordinate of the intersection of a line originating at (0,0,0) in direction of `fittingVector` with the triangle defined by vertices (0,0,1),(0,1,0),(1,0,0)